design/pipeline-triggers.md

```diff
@@ -0,0 +1,369 @@
+# Triggers in pipelines
+
+Triggers are events on which you can start your pipeline run automatically. You can enable triggers on your pipeline by subscribing to both internal and external events.
+
+At high level there are 3 different types of pipeline triggers.
+1. Resource triggers
+2. Schedule triggers
+3. Webhook triggers
+
+## Resource triggers
+You can enable triggers on the resources defined in your pipeline. Resources can be of types pipelines, repositories, containers and packages.
+Triggers are enabled by default on all the resources. However, you can choose to override/disable triggers for each resource.
+
+
+
+### Pipelines
+A new pipeline is triggered automatically whenever a new run of the `pipeline` resource is succesfully completed. See [pipeline resources](pipeline-resources.md#resources-pipelines) for more details.
+
+#### Scenarios
+- I would like to trigger my pipeline when a new ‘helm-package’ artifact is published by ‘Helm-CI’ pipeline that ran on `releases/*` branch.
+- I would like to trigger my pipeline when a new ‘helm-package’ artifact is published and tested as part of Helm-CI pipeline and tagged as 'Production'.
+- I would like to trigger my pipeline when ‘TFS-Update’ pipeline has completed ‘Ring2’ stage.  
+
+#### Schema
+```yaml
+resources:       
+  pipelines:
+  - pipeline: string 
+    source: string  
+    trigger:     # Optional; Triggers are enabled by default.
+      branches:  # branch conditions to filter the events, optional; Defaults to all branches.
+        include: [ string ]  # branches to consider the trigger events, optional; Defaults to all branches.
+        exclude: [ string ]  # branches to discard the trigger events, optional; Defaults to none.
+      stages: [ string ]  # trigger after completion of given stage, optional; Defaults to all stage completion.
+      tags: [ string ]  # tags on which the trigger events are considered, optional; Defaults to any tag or no tag.
+```
+
+#### Examples
+Triggers are enabled by default unless expliciltly opted out. You can disable the triggers on the `pipeline` resource.
```

Comment:
I agree that for pipelines, automatic triggers make the most sense.

Reply:
Even for failed pipelines? I would default to succeeded or succeeded with issues.

Reply:
(which matches the default condition on jobs/stages)

Reply:
And how do you explicitly filter by result? Or are successful pipelines only supported for chaining?

Reply:
Only the successful or partially succeeded pipelines are considered for triggers in this scope. We do the same in RM today. 

If we find valid scenarios for piplines with failed (jobs/stages), we can provide that support later on.